### PR TITLE
Break dependency on k/k and remove replace statements in go.mod file

### PR DESCRIPTION
**Pull request by amacaskill**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
> /kind bug

/kind cleanup

> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**:
This PR allows us to remove most of the replace statements in go.mod. The client-go replace statement was still needed in order for go mod tidy to succeed. See details [here](https://github.com/kubernetes-sigs/gcp-compute-persistent-disk-csi-driver/pull/1169#discussion_r1131537873)

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
Break dependency on k/k and remove replace statements in go.mod file.
```

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -113,33 +113,4 @@ require (
...
-	k8s.io/legacy-cloud-providers => k8s.io/legacy-cloud-providers v0.26.0
-	k8s.io/metrics => k8s.io/metrics v0.26.0
-	k8s.io/mount-utils => k8s.io/mount-utils v0.26.0
-	k8s.io/node-api => k8s.io/node-api v0.26.0
-	k8s.io/pod-security-admission => k8s.io/pod-security-admission v0.26.0
-	k8s.io/sample-apiserver => k8s.io/sample-apiserver v0.26.0
-	k8s.io/sample-cli-plugin => k8s.io/sample-cli-plugin v0.26.0
-	k8s.io/sample-controller => k8s.io/sample-controller v0.26.0
-)
+replace k8s.io/client-go => k8s.io/client-go v0.26.0
```
Comment (msau42) [inline go.mod:116]: I think even this can be removed

Comment (msau42) [reply go.mod:116]: Can you try setting 0.26.0 in L32?

Comment (amacaskill) [reply go.mod:116]: Oh I think that worked. I'll try to remove that for PDCSI also.

Comment (amacaskill) [reply go.mod:116]: Changing to 0.26.0 and removing the replace statement results in 

```
go: sigs.k8s.io/gcp-filestore-csi-driver/pkg/client/clientset/versioned imports
        k8s.io/client-go/util/flowcontrol imports
        k8s.io/apimachinery/pkg/util/clock: module k8s.io/apimachinery@latest found (v0.31.0), but does not contain package k8s.io/apimachinery/pkg/util/clock
go: sigs.k8s.io/gcp-filestore-csi-driver/pkg/csi_driver imports
        k8s.io/client-go/informers imports
        k8s.io/api/auditregistration/v1alpha1: module k8s.io/api@latest found (v0.31.0), but does not contain package k8s.io/api/auditregistration/v1alpha1
go: sigs.k8s.io/gcp-filestore-csi-driver/pkg/csi_driver imports
        k8s.io/client-go/informers imports
        k8s.io/api/batch/v2alpha1: module k8s.io/api@latest found (v0.31.0), but does not contain package k8s.io/api/batch/v2alpha1
go: sigs.k8s.io/gcp-filestore-csi-driver/pkg/csi_driver imports
        k8s.io/client-go/informers imports
        k8s.io/api/settings/v1alpha1: module k8s.io/api@latest found (v0.31.0), but does not contain package k8s.io/api/settings/v1alpha1
go: sigs.k8s.io/gcp-filestore-csi-driver/pkg/webhook imports
        sigs.k8s.io/controller-runtime/pkg/certwatcher tested by
        sigs.k8s.io/controller-runtime/pkg/certwatcher.test imports
        sigs.k8s.io/controller-runtime imports
        sigs.k8s.io/controller-runtime/pkg/builder imports
        sigs.k8s.io/controller-runtime/pkg/client imports
        k8s.io/client-go/metadata: package k8s.io/client-go/metadata provided by k8s.io/client-go at latest version v0.31.0 but not at required version v11.0.1-0.20190805182717-6502b5e7b1b5+incompatible
```

I also tried v0.31.0, but it keeps saying it wants k8s.io/client-go/metadata at v11.0.1-0.20190805182717-6502b5e7b1b5+incompatible

Comment (msau42) [reply go.mod:116]: bummer, this is still a good improvement though so let's go with what we have for now

**Pull request merged**